### PR TITLE
Feat/add file picker

### PR DESCRIPTION
**Pull request by Merkur39**
Add FilePicker so the user can choose a file directly from their Drive

### CODE REVIEW

File: src/assets/icons/MoveTo.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M5.509 1C5.78017 1 6.15694 1.15694 6.34837 1.34837L7.65163 2.65163C7.84403 2.84403 8.2308 3 8.50044 3H15.003C15.5536 3 16 3.45576 16 4.00247V12.9915C16 14.1007 15.1055 15 14.0059 15H1.99406C0.892771 15 0 14.1029 0 12.9941V2.00685C0 1.45078 0.450975 1 0.990778 1H5.509ZM11.5 6C11.1095 5.60948 10.4763 5.60948 10.0858 6C9.7253 6.36048 9.69757 6.92771 10.0026 7.32001L10.0858 7.41421L10.5858 8H6C5.44772 8 5 8.44771 5 9C5 9.51284 5.38604 9.93551 5.88338 9.99327L6 10H10.5858L10.0858 10.5858C9.69526 10.9763 9.69526 11.6095 10.0858 12C10.4463 12.3605 11.0135 12.3882 11.4058 12.0832L11.5 12L13.7071 9.70711C14.0676 9.34662 14.0953 8.77939 13.7903 8.3871L13.7071 8.29289L11.5 6Z" fill="#1D212A" fill-opacity="0.72"/>
+</svg>
```
Comment (Crash--) [inline src/assets/icons/MoveTo.svg:3]: @joel-costa Should we remove the fill and opacity attribute? How do we display it in other context with other colors configuration ? 

Also should it be added to UI ? Should we use it somewhere else in Cozy ?

File: src/components/ModelSteps/Scan.jsx [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
+    []
+  )
+
+  const onPickFile = useCallback(() => {
+    setIsPickFileModalOpen(true)
+  }, [])
+
+  const onFileChange = useCallback(file => {
+    file && setFile(file)
+  }, [])
```
Comment (JF-Cozy) [inline src/components/ModelSteps/Scan.jsx:61]: nit: file n'est pas censé être en dépendance ?

File: src/components/ModelSteps/Scan.jsx [hunk 3]
```diff
@@ -1,23 +1,82 @@
...
+  const handleChangeFilePicker = useCallback(fileId => {
+    setFileId(fileId)
+  }, [])
```
Comment (JF-Cozy) [inline src/components/ModelSteps/Scan.jsx:64]: nit : fileId n'est pas censé être en dépendance ?

File: src/utils/fetchBlobFile.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { FILES_DOCTYPE } from 'src/doctypes'
+
+/**
+ * @param {CozyClient} client - Instance of CozyClient
+ * @param {IOCozyFile} fileId - Id of file
+ * @returns {Promise<Blob>}
+ */
+export const fetchBlobFile = async (client, fileId) => {
```
Comment (JF-Cozy) [inline src/utils/fetchBlobFile.js:8]: nit: est-ce qu'on pourrait le mettre dans cozy-client ?

File: src/assets/icons/MoveTo.svg (hunk 1 above)
Comment (JF-Cozy) [reply src/assets/icons/MoveTo.svg:3]: +1 pour les fill, voir la doc de cozy-ui pour l'ajout d'icone.

Comment (JF-Cozy) [reply src/assets/icons/MoveTo.svg:3]: +1 pour les fill, voir la doc de cozy-ui pour l'ajout d'icone.

File: src/components/ModelSteps/Scan.jsx (hunk 2 above)
Comment (Merkur39) [reply src/components/ModelSteps/Scan.jsx:61]: Non, car ici on ne parle pas de ce `file` (ligne 37) : `const [file, setFile] = useState(null)`, mais bien du paramètre passé à la fonction

File: src/components/ModelSteps/Scan.jsx (hunk 3 above)
Comment (Merkur39) [reply src/components/ModelSteps/Scan.jsx:64]: Même raison que ci-dessus (avec un autre state)

File: src/utils/fetchBlobFile.js (hunk 4 above)
Comment (Merkur39) [reply src/utils/fetchBlobFile.js:8]: En effet, c'est en cours 👍 
https://github.com/cozy/cozy-client/pull/1109

**Comment by Merkur39** [comment]
[Brancher le file picker](https://trello.com/c/cTYQYgVQ/828-brancher-le-file-picker)

File: src/assets/icons/MoveTo.svg (hunk 1 above)
Comment (joel-costa) [reply src/assets/icons/MoveTo.svg:3]: +1 to remove the fill ans opacity attribute

Currently it is not used somewhere else in Cozy but I think it will be in the future, probably in a connector to change the destination folder

Comment (Merkur39) [reply src/assets/icons/MoveTo.svg:3]: Ok, icon added with this PR: https://github.com/cozy/cozy-ui/pull/2027

File: src/components/ModelSteps/Scan.jsx (hunk 2 above)
Comment (Ldoppea) [reply src/components/ModelSteps/Scan.jsx:60]: Ca pourrait être bien de renommer l'argument de la fonction en autre chose que `file` du coup, c'est source de confusion

Comment (Ldoppea) [reply src/components/ModelSteps/Scan.jsx:60]: ex: `fileToSet` ou `newFile`

File: src/components/ModelSteps/Scan.jsx [hunk 5]
```diff
@@ -1,23 +1,86 @@
...
+
+  const openPickerFileModal = useCallback(() => {
+    setIsFilePickerOpen(true)
+  }, [])
+  const closePickFileModal = useCallback(
+    () => setIsPickFileModalOpen(false),
+    []
+  )
+
+  const onPickFile = useCallback(() => {
```
Comment (Ldoppea) [inline src/components/ModelSteps/Scan.jsx]: nit: I expected this method to be called when the user picks a file, not when they click on "I already have a picture". Maybe this method name can be improved?

File: src/components/ModelSteps/Scan.jsx [hunk 6]
```diff
@@ -1,23 +1,86 @@
...
   const { alreadyScan } = useStepperDialog()
 
-  const onFileChange = file => file && setFile(file)
+  const [isFilePickerOpen, setIsFilePickerOpen] = useState(false)
+  const [fileId, setFileId] = useState('')
+  const [isPickFileModalOpen, setIsPickFileModalOpen] = useState(false)
+
+  const closeFilePicker = useCallback(() => setIsFilePickerOpen(false), [])
+
+  const openPickerFileModal = useCallback(() => {
```
Comment (Ldoppea) [inline src/components/ModelSteps/Scan.jsx]: `closeFilePicker` and `openPickerFileModal` should be named using same convention

Comment (Ldoppea) [reply src/components/ModelSteps/Scan.jsx]: More globally it is hard to follow why there are some methods with "modal" and some other don't.

closeFilePicker calls setIsFilePickerOpen
openPickerFile**Modal** calls setIsFilePickerOpen
onPickFile calls setIsPickFile**Modal**Open
onFileChange calls setIsPickFile**Modal**Open

File: src/locales/en.json [hunk 7]
```diff
@@ -217,6 +217,11 @@
     "title": "New paper%{name}"
   },
   "Scan": {
+    "modal": {
+      "cozySelect": "Select in my Cozy",
```
Comment (Ldoppea) [inline src/locales/en.json:221]: I'm not sure about this one. Maybe `Select from my cozy`? We should google that.

**Review by Ldoppea — APPROVED** [review]
I'm pretty sure a lot of this complexity may be moved into cozy-ui FilePicker (maybe in a hook):
- handleChangeFilePicker
- onFileChange
- closeFilePicker
- openPickerFileModal
- setIsFilePickerOpen
- setIsPickFileModalOpen

Something like `const {fileId} = useFilePicker()`

File: src/components/ModelSteps/Scan.jsx (hunk 6 above)
Comment (Ldoppea) [reply src/components/ModelSteps/Scan.jsx]: What is the difference between `setIsPickFileModalOpen` and `setIsFilePickerOpen`?

Comment (Merkur39) [reply src/components/ModelSteps/Scan.jsx]: I rename `setIsPickFileModalOpen ` to `setIsImportFileModalOpen`
- `setIsImportFileModalOpen` open modal/bottomDrawer menu
- `setIsFilePickerOpen` open FilePicker

**Comment by Crash--** [comment]
@Ldoppea definitely! @Merkur39 can you create an issue on cozy-ui to not lose it?

**Comment by Merkur39** [comment]
@Ldoppea It's not so complex, since some functions have been renamed for a better understanding.
- ~~handleChangeFilePicker~~ => onChangeFilePicker
- ~~closeFilePicker~~ => closeFilePickerModal
- ~~openPickerFileModal~~ => openFilePickerModal
- ~~setIsFilePickerOpen~~ => setIsFilePickerModalOpen
- ~~setIsPickFileModalOpen~~ => setIsImportFileModalOpen

And only these functions are for the FilePicker:
- onChangeFilePicker
- closeFilePickerModal

The others are to manage either the previous ActionMenu or the FilePicker display.

I don't understand the use/implementation of a Hook in this case.

**Comment by Ldoppea** [comment]
"Complexity" does not always mean "complex" 😄 

What I meant is that everytime we will need a file picker we will have to implement those methods (even if they are simple) to handle the popup lifecycle. So we can simplify the API by moving them in the component.
The consuming app doesn't really care if the modal is opened or not, what is important is to know the fileId and to react to its changes.

The hook was just an exemple without any reflexion behind it, maybe the solution will not involve any hook. So don't mind about it 😅

**Comment by Merkur39** [comment]
Ok I understand. There is indeed an improvement to make but more global I think.
I think for example of the `ContactsListModal` component which works the same way

It would be nice to talk about this at our next "front" meeting

**Comment by Crash--** [comment]
@Merkur39 let's create an issue on UI's side and let's talk about that there. No need to wait our weekly meeting to open the discussion. Meeting should be there to take a decision if needed, not to think about the solution ;)
